lib/sidekiq/instrument/middleware/server.rb

```diff
@@ -17,15 +17,21 @@ def call(worker, job, _queue, &block)
       execution_time_ms = (Time.now - start_time) * 1000
       Statter.statsd.measure(metric_name(worker, 'runtime'), execution_time_ms)
       Statter.dogstatsd&.timing('sidekiq.runtime', execution_time_ms, worker_dog_options(worker))
-    rescue StandardError => e
+    rescue Exception => e
+      dd_options = worker_dog_options(worker)
+      dd_options[:tags] << "error:#{e.class.name}"
+
       # if we have retries left, increment the enqueue.retry counter to indicate the job is going back on the queue
       if max_retries(worker) > current_retries(job) + 1
         WorkerMetrics.trace_workers_increment_counter(worker.class.to_s.underscore)
-        Statter.dogstatsd&.increment('sidekiq.enqueue.retry', worker_dog_options(worker))
+        Statter.dogstatsd&.increment('sidekiq.enqueue.retry', dd_options)
       end
 
+      # categorize rate limit lock errors differently from actual errors
+      error_string = e.class.name.eql?("RedisRateLimit::Throttle::LockFailedError") ? 'sidekiq.error.redis_rate_lock' : 'sidekiq.error'
```

Comment:
`e.is_a?(RedisRateLimit::Throttle::LockFailedError)` doesn't work here either? because it's an internal class?

Reply:
We cannot make open-source code dependent on internal code. We'll have to regroup on this.

Reply:
What if you add the error class as a tag instead?

Reply:
In fact, you do that already, so why make this differentiation?

Reply:
> We cannot make open-source code dependent on internal code. We'll have to regroup on this.

I understand that it's making it dependent on internal code, but it's also not interfering with anything using this gem that doesn't have any errors of that type. 

> What if you add the error class as a tag instead?

We're already doing that in this PR: https://github.com/enova/sidekiq-instrument/pull/32/files#diff-49e482f9b1d7c73d1688382594ad4ac70b7c7d9b439465d19804de0de86bc69eR21-R22

Reply:
Yeah, so why make this differentiation then?